bundles/model/org.eclipse.smarthome.model.rule.tests/src/org/eclipse/smarthome/model/rule/runtime/internal/engine/RuleExecutionTest.java

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.model.rule.runtime.internal.engine;
+
+import static org.junit.Assert.*;
+
+import java.io.ByteArrayInputStream;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.events.Event;
+import org.eclipse.smarthome.core.events.EventFilter;
+import org.eclipse.smarthome.core.events.EventPublisher;
+import org.eclipse.smarthome.core.events.EventSubscriber;
+import org.eclipse.smarthome.core.items.GenericItem;
+import org.eclipse.smarthome.core.items.GroupItem;
+import org.eclipse.smarthome.core.items.ItemRegistry;
+import org.eclipse.smarthome.core.items.events.ItemCommandEvent;
+import org.eclipse.smarthome.core.items.events.ItemEventFactory;
+import org.eclipse.smarthome.core.library.CoreItemFactory;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.model.core.ModelRepository;
+import org.eclipse.smarthome.model.rule.rules.Rule;
+import org.eclipse.smarthome.model.rule.runtime.RuleEngine;
+import org.eclipse.smarthome.model.rule.runtime.internal.engine.RuleTriggerManager.TriggerTypes;
+import org.eclipse.smarthome.test.java.JavaOSGiTest;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * @author Kai Kreuzer - initial contribution and API.
+ */
+public class RuleExecutionTest extends JavaOSGiTest {
+
+    private final static String TESTMODEL_NAME = "testModel.rules";
+    private ModelRepository modelRepository;
+    private ItemRegistry itemRegistry;
+    private EventPublisher eventPublisher;
+    private Event resultEvent;
+
+    private final EventSubscriber eventSubscriber = new EventSubscriber() {
```

Comment:
Wouldn't this rather be an excellent candidate for a mock? You could grab the expected event with an `ArgumentCaptor`...

Reply:
I'd prefer to leave it as it is for now - feel free to refactor it later on, if you think it becomes more readable with a mock.